spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/utils/jackson/InstanceEventMixin.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2014-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.codecentric.boot.admin.server.utils.jackson;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import de.codecentric.boot.admin.server.domain.events.InstanceDeregisteredEvent;
+import de.codecentric.boot.admin.server.domain.events.InstanceEndpointsDetectedEvent;
+import de.codecentric.boot.admin.server.domain.events.InstanceEvent;
+import de.codecentric.boot.admin.server.domain.events.InstanceInfoChangedEvent;
+import de.codecentric.boot.admin.server.domain.events.InstanceRegisteredEvent;
+import de.codecentric.boot.admin.server.domain.events.InstanceRegistrationUpdatedEvent;
+import de.codecentric.boot.admin.server.domain.events.InstanceStatusChangedEvent;
+
+/**
+ * Jackson Mixin class helps in serialize/deserialize {@link InstanceEvent}s.
+ *
+ * @author Stefan Rempfer
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "type")
+@JsonSubTypes({ @JsonSubTypes.Type(value = InstanceEndpointsDetectedEvent.class, name = "ENDPOINTS_DETECTED"),
+		@JsonSubTypes.Type(value = InstanceRegistrationUpdatedEvent.class, name = "REGISTRATION_UPDATED"),
+		@JsonSubTypes.Type(value = InstanceInfoChangedEvent.class, name = "INFO_CHANGED"),
+		@JsonSubTypes.Type(value = InstanceDeregisteredEvent.class, name = "DEREGISTERED"),
+		@JsonSubTypes.Type(value = InstanceRegisteredEvent.class, name = "REGISTERED"),
+		@JsonSubTypes.Type(value = InstanceStatusChangedEvent.class, name = "STATUS_CHANGED") })
```

Comment:
I'd suggest to introduce a a constent for the type. E.g.
`@JsonSubTypes.Type(value = InstanceStatusChangedEvent.class, name = InstanceStatusChangedEvent.TYPE) `